webserver/src/main/java/io/kestra/webserver/controllers/ExecutionController.java

```diff
@@ -799,38 +817,38 @@ public static class StateRequest {
     }
 
     @ExecuteOn(TaskExecutors.IO)
-    @Delete(uri = "/{executionId}/kill")
+    @Delete(uri = "/{executionId}/kill{?isOnKillCascade}", produces = MediaType.TEXT_JSON)
     @Operation(tags = {"Executions"}, summary = "Kill an execution")
-    @ApiResponse(responseCode = "204", description = "On success")
+    @ApiResponse(responseCode = "202", description = "Execution kill was requested successfully")
     @ApiResponse(responseCode = "409", description = "if the executions is already finished")
     @ApiResponse(responseCode = "404", description = "if the executions is not found")
     public HttpResponse<?> kill(
-        @Parameter(description = "The execution id") @PathVariable String executionId
+        @Parameter(description = "The execution id") @PathVariable String executionId,
+        @Parameter(description = "Specifies whether killing the execution also kill all subflow executions.") @QueryValue(defaultValue = "true") Boolean isOnKillCascade
     ) throws InternalException {
+
         Optional<Execution> maybeExecution = executionRepository.findById(tenantService.resolveTenant(), executionId);
         if (maybeExecution.isEmpty()) {
             return HttpResponse.notFound();
         }
 
         var execution = maybeExecution.get();
-        if (execution.getState().isTerminated()) {
-            throw new IllegalStateException("Execution is already finished, can't kill it");
-        }
 
-        if (execution.getState().isPaused()) {
-            // Must be resumed and killed, no need to send killing event to the worker as the execution is not executing anything in it.
-            // An edge case can exist where the execution is resumed automatically before we resume it with a killing.
-            this.executionService.resume(execution, State.Type.KILLING);
-            return HttpResponse.noContent();
```

Comment:
You remove resuming paused execution from here but I didn't see it inside the executor so it may be missing.

Reply:
Because, the Executor is now fully responsible for handling ExecutionKill this check is no more required as already performed by the `ExecutionService`: https://github.com/kestra-io/kestra/blob/c6ee44568ccc53ea5c905fc5aa14300319186568/core/src/main/java/io/kestra/core/services/ExecutionService.java#L360-L376